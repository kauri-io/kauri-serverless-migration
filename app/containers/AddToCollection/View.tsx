import AlertView from '../../components/Modal/AlertView'
import AddToCollectionModalContent from './AddToCollectionModalContent'
import { Query } from 'react-apollo'
import { path } from 'ramda'
import { ICollection } from './CollectionsContent'
import { ISection } from './SectionsContent'
import Loading from '../../components/Loading'
import { ErrorMessage } from '../../lib/with-apollo-error'
import { Label } from '../..//components/Typography'
import { getCollectionsForUser as query } from '../../queries/Collection'
import {
    getCollectionsForUser,
    getCollectionsForUserVariables,
} from '../../queries/__generated__/getCollectionsForUser'
import { useState } from 'react'
import { addResourceToCollectionAction } from './Module'
import { ResourceTypeInput } from '../../__generated__/globalTypes'

interface IChosen {
    chosenCollection: ICollection | null
    chosenSection: ISection | null
}

interface IProps {
    closeModalAction: () => void
    userId: string | null
    routeChangeAction: (route: string) => void
    resourceId: string
    type: ResourceTypeInput
    addResourceToCollectionAction: typeof addResourceToCollectionAction
}

const AddToCollectionComponent: React.FunctionComponent<IProps> = ({
    closeModalAction,
    userId,
    routeChangeAction,
    resourceId,
    type,
    addResourceToCollectionAction,
}) => {
    const [state, setState] = useState<IChosen>({
        chosenCollection: null,
        chosenSection: null,
    })

    return (
        <Query<getCollectionsForUser, getCollectionsForUserVariables>
            query={query}
            variables={{ filter: { ownerIdEquals: userId } }}
        >
            {props => {
                if (props.loading) {
                    return <Loading />
                }
                if (props.error) {
                    return (
                        <ErrorMessage
                            data={{ error: { message: props.error.message } }}
                        />
                    )
                }

                if (
                    props.data &&
                    props.data.searchCollections &&
                    Array.isArray(props.data.searchCollections.content)
                ) {
                    const collections = path<ICollection[]>([
                        'searchCollections',
                        'content',
                    ])(props.data)

                    const collectionsThatDoNotHaveTheChosenArticleId =
                        (collections &&
                            Array.isArray(collections) &&
                            collections.filter(collection => {
                                // Filter out collections that have the resourceId already
                                if (
                                    collection.sections.find(
                                        section =>
                                            !!section.resources.find(
                                                resource =>
                                                    resource.id === resourceId
                                            )
                                    )
                                ) {
                                    return null
                                } else {
                                    return collection
                                }
                            })) ||
                        []

                    const articleAlreadyInAllCollections =
                        collectionsThatDoNotHaveTheChosenArticleId &&
                        collectionsThatDoNotHaveTheChosenArticleId.length === 0

                    return Array.isArray(collections) &&
                        collections.length > 0 ? (
                        <AlertView
                            closeModalAction={closeModalAction}
                            confirmButtonText={
                                articleAlreadyInAllCollections
                                    ? 'CREATE NEW COLLECTION'
                                    : 'CONFIRM'
                            }
                            confirmButtonAction={() => {
                                if (articleAlreadyInAllCollections) {
                                    closeModalAction()
                                    routeChangeAction(
                                        `/create-collection?resourceId=${resourceId}&type=${type}`
                                    )
                                }

                                if (
                                    state.chosenCollection &&
                                    state.chosenSection
                                ) {
                                    const chosenSectionid =
                                        state.chosenSection.id
                                    const chosenCollectionSections = state.chosenCollection.sections.find(
                                        section =>
                                            section.id === chosenSectionid
                                    )

                                    // Always insert article at the end
                                    const position = chosenCollectionSections
                                        ? Array.isArray(
                                              chosenCollectionSections.resources
                                          )
                                            ? chosenCollectionSections.resources
                                                  .length
                                            : 0
                                        : 0

                                    addResourceToCollectionAction(
                                        {
                                            id: state.chosenCollection.id,
                                            position,
                                            resourceId: {
                                                id: resourceId,
                                                type,
                                            },
                                            sectionId: state.chosenSection
                                                ? state.chosenSection.id
                                                : '',
                                            closeModalAction,
                                            routeChangeAction,
                                        },
                                        closeModalAction
                                    )
                                }
                            }}
                            content={
                                <AddToCollectionModalContent
                                    changeToPrefilledArticleCreateCollectionRoute={() => {
                                        closeModalAction()
                                        routeChangeAction(
                                            `/create-collection?resourceId=${resourceId}&type=${type}`
                                        )
                                    }}
                                    articleAlreadyInAllCollections={
                                        !!articleAlreadyInAllCollections
                                    }
                                    collectionsThatDoNotHaveTheChosenArticleId={
                                        collectionsThatDoNotHaveTheChosenArticleId
                                    }
                                    parentState={state}
                                    setCollection={({ chosenCollection }) =>
                                        setState({
                                            chosenCollection,
                                            chosenSection: null,
                                        })
                                    }
                                    setSection={({ chosenSection }) =>
                                        setState({ ...state, chosenSection })
                                    }
                                />
                            }
                            title={'Add to Collection'}
                        />
                    ) : (
                        <AlertView
                            title={'Create new collection'}
                            closeModalAction={closeModalAction}
                            confirmButtonAction={() => {
                                closeModalAction()
                                routeChangeAction(
                                    `/create-collection?resourceId=${resourceId}&type=${type}`
                                )
                            }}
                            confirmButtonText={'Create new collection'}
                            content={
                                <section>
                                    <Label>
                                        You don't have any collections.
                                    </Label>
                                </section>
                            }
                        />
                    )
                }
            }}
        </Query>
    )
}

export default AddToCollectionComponent
