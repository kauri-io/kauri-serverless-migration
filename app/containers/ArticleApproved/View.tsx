import React from 'react'
import Head from 'next/head'
import { cond, equals, always, tail } from 'ramda'
import styled from 'styled-components'
import ArticleCard from '../../components/Card/ArticleCardMaterial'
import PrimaryButton from '../../components/Button/PrimaryButton'
import { Title1, BodyCard } from '../../components/Typography'
import { Article } from '../../queries/Fragments/__generated__/Article'
import { ShareButtons } from '../../components/Tooltip/ShareButtons'
import getArticleHref from '../../lib/getArticleHref'

export interface IProps {
    data: {
        getArticle: Article
    }
    hostName: string
    routeChangeAction: (route: string) => void
    type: 'published' | 'approved' | 'drafted' | 'updated'
    user: { id: string }
    isLoggedIn: boolean
}

const Container = styled.section`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: calc(100vh - 76px);
    background-color: ${props => props.theme.primaryTextColor};
    > :nth-child(2) {
        margin-bottom: ${props => props.theme.space[3]}px;
    }
`

const DescriptionContainer = styled.div`
    max-width: 500px;
    text-align: center;
    > * {
        word-break: break-word;
    }
`

const ArticleApprovedActionButtons = styled.div`
    display: flex;
    > :first-child:not(:only-child) {
        margin-right: 10px;
    }
    > :only-child {
        margin-right: 0px;
    }
    margin-top: 20px;
    > :first-child {
        margin-right: 0px;
    }
`

class ArticleApproved extends React.Component<IProps> {
    render() {
        const { data, routeChangeAction, type, hostName } = this.props
        const subjectCopy = cond<string, string>([
            [
                equals('updated'),
                always(
                    'draft has been updated. You can view all drafts on your profile page.'
                ),
            ],
            [
                equals('drafted'),
                always(
                    `has been saved as a draft.
           You can view all drafts on your profile page.
           All drafts are unlisted, rather than private.
           This means you can send the link to someone and they can view it, but your article will not be discoverable or searchable until it is published.
          `
                ),
            ],
            [equals('published'), always('is now live')],
            [
                equals('approved'),
                always('now needs publishing from author before going live'),
            ],
        ])(type)

        const capitalize = (stringToCapitalize: string) =>
            stringToCapitalize.length > 0 &&
            stringToCapitalize[0].toUpperCase() + tail(stringToCapitalize)

        if (this.props.data.getArticle) {
            const article = data.getArticle

            return (
                <Container>
                    <Head>
                        <title>{`Kauri - Article ${capitalize(type)}`}</title>
                    </Head>
                    <Title1 color="white">{capitalize(type)}</Title1>
                    <DescriptionContainer>
                        <BodyCard color="white">{`The article ${subjectCopy}`}</BodyCard>
                    </DescriptionContainer>
                    <ArticleCard href={getArticleHref(article)} {...article} />
                    <ArticleApprovedActionButtons>
                        <ShareButtons
                            horizontal={true}
                            title={String(article.title)}
                            url={String(
                                `https://${hostName.replace(
                                    'api.',
                                    ''
                                )}/article/${article.id}?utm_campaign=published`
                            )}
                        />
                        <PrimaryButton
                            onClick={() =>
                                routeChangeAction(
                                    `/public-profile/${this.props.user &&
                                        this.props.user.id}`
                                )
                            }
                        >
                            My Articles
                        </PrimaryButton>
                    </ArticleApprovedActionButtons>
                </Container>
            )
        }
    }
}

export default ArticleApproved
