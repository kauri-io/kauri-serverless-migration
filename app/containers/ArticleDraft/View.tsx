import DraftArticleHeader from "./DraftArticleHeader";
import DraftArticleContent from "./DraftArticleContent";
import { IOption } from "../../common/PublishingSelector";

interface IProps {
  data: {
    getArticle: any;
  };
  deleteDraftArticleAction: (
    { id, version }: { id: string; version: number }
  ) => void;
  openModalAction: (children?: any) => void;
  closeModalAction: () => void;
  publishArticleAction: any;
  userId: string;
  routeChangeAction: (route: string) => void;
  communities: Array<{ community: IOption }>;
}
export default ({
  data: { getArticle },
  deleteDraftArticleAction,
  closeModalAction,
  publishArticleAction,
  openModalAction,
  userId,
  routeChangeAction,
  communities,
}: IProps) => (
  <>
    <DraftArticleHeader
      {...getArticle}
      communities={communities}
      userId={userId}
      routeChangeAction={routeChangeAction}
      publishArticleAction={publishArticleAction}
      openModalAction={openModalAction}
      closeModalAction={closeModalAction}
    />
    <DraftArticleContent
      {...getArticle}
      closeModalAction={closeModalAction}
      openModalAction={openModalAction}
      deleteDraftArticleAction={deleteDraftArticleAction}
    />
  </>
);
