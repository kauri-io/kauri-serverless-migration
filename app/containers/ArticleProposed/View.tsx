import React from 'react'
import styled from 'styled-components'
import Head from 'next/head'
import ArticleCard from '../../components/Card/ArticleCardMaterial'
import PrimaryButton from '../../components/Button/PrimaryButton'
import { Title2, BodyCard } from '../../components/Typography'
import { getArticleURL } from '../../lib/getURLs'

const Container = styled.section`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100%;
    background-color: ${props => props.theme.primaryTextColor};
    > :first-child {
        margin-bottom: 30px;
    }
`

const SectionContainer = styled.div`
    margin-top: ${props => props.theme.space[3]}px;
    & > h2 {
        margin-bottom: ${props => props.theme.space[1]}px;
    }
`

const ArticleProposed = ({
    data: { getArticle },
    routeChangeAction,
    userId,
}) => {
    return (
        <Container>
            <Head>
                <title>{'Kauri - Article Proposed'}</title>
            </Head>
            <SectionContainer>
                <Title2 color="white">Article Update Submitted</Title2>
                <BodyCard color="white">
                    Waiting for original author approval
                </BodyCard>
            </SectionContainer>
            <ArticleCard href={getArticleURL(getArticle)} {...getArticle} />
            <SectionContainer>
                <PrimaryButton
                    onClick={() =>
                        routeChangeAction(`/public-profile/${userId}`)
                    }
                >
                    My Articles
                </PrimaryButton>
            </SectionContainer>
        </Container>
    )
}

export default ArticleProposed
