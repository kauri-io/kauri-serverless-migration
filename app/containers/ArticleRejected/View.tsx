import React from 'react'
import styled from 'styled-components'
import SecondaryButton from '../../components/Button/SecondaryButton'
import { H2 } from '../../components/Typography'
// import { ArticleApprovedConfirmationLogoBadge } from "../../common/ActionBadge";

const ConfirmationSubject = styled.h2`
    color: #ffffff;
    font-weight: normal;
    font-size: 16px;
    margin-bottom: 50px;
`

const Container = styled.section`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100%;
    background-color: ${props => props.theme.primaryTextColor};
    > :first-child {
        margin-bottom: 30px;
        margin-bottom: 7px;
    }
`

interface IProps {
    data: any
    routeChangeAction: any
    userId: any
}

const ArticleRejected = (props: IProps) => {
    const { data, routeChangeAction, userId } = props
    return (
        <Container>
            <H2>Article Rejected</H2>
            <ConfirmationSubject>
                {(data && data.getArticle && data.getArticle.subject) ||
                    'The author has been notified by email with your note.'}
            </ConfirmationSubject>
            <SecondaryButton
                onClick={() => routeChangeAction(`/public-profile/${userId}`)}
            >
                Back to my articles
            </SecondaryButton>
        </Container>
    )
}

export default ArticleRejected
