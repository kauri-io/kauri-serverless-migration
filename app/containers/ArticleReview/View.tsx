import React from "react";
import styled from "../../../lib/styled-components";
import Header from "./Header";
import Diffs from "../../../../kauri-components/components/DiffViewer";
import { Label } from "../../../../kauri-components/components/Typography";
import ScrollIndicator from "../../../../kauri-components/components/ScrollIndicator";
import Loading from "../../common/Loading";
import theme from "../../../../kauri-components/lib/theme-config";
import { ICommunities } from "../../../lib/Module";

const DEFAULT_CARD_WIDTH = theme.DEFAULT_CARD_WIDTH;

const Container = styled.div`
  background: white;
`;

const Content = styled.div`
  display: flex;
  flex-direction: row;
  padding: ${props => props.theme.space[3]}px ${props => props.theme.padding};
`;

const Details = styled.div`
  display: flex;
  flex-direction: column;
  max-width: ${DEFAULT_CARD_WIDTH}px;
  margin-left: ${props => props.theme.space[3]}px;
`;

interface IProps {
  userId: string;
  id: string;
  version: string;
  communities: ICommunities;
  CurrentArticle: any;
  ProposedUpdate: any;
  routeChangeAction: (route: string) => void;
  openModalAction: (children: any) => void;
  closeModalAction: () => void;
  rejectArticleAction: (
    { cause, id, version }: { cause: string; id: string; version: number }
  ) => void;
  approveArticleAction: (
    {
      id,
      version,
      contentHash,
      author,
      dateCreated,
    }: {
      id: string;
      version: number;
      contentHash: string;
      author: string;
      dateCreated: string;
    }
  ) => void;
}

class ArticleReviewView extends React.Component<IProps, {}> {
  render() {
    const { CurrentArticle, ProposedUpdate, communities } = this.props;
    if (!CurrentArticle.getArticle || !ProposedUpdate.getArticle) {
      return <Loading />;
    }

    const current = {
      attributes: CurrentArticle.getArticle.attributes,
      author: CurrentArticle.getArticle.author,
      content: JSON.parse(CurrentArticle.getArticle.content).markdown,
      contentHash: CurrentArticle.getArticle.contentHash,
      dateCreated: CurrentArticle.getArticle.dateCreated,
      owner: CurrentArticle.getArticle.owner.id,
      tags: CurrentArticle.getArticle.tags,
      title: CurrentArticle.getArticle.title,
      version: CurrentArticle.getArticle.version,
    };
    const proposed = {
      attributes: ProposedUpdate.getArticle.attributes,
      author: ProposedUpdate.getArticle.author,
      content: JSON.parse(ProposedUpdate.getArticle.content).markdown,
      contentHash: ProposedUpdate.getArticle.contentHash,
      dateCreated: ProposedUpdate.getArticle.dateCreated,
      status: ProposedUpdate.getArticle.status,
      tags: ProposedUpdate.getArticle.tags,
      title: ProposedUpdate.getArticle.title,
      updateComment: ProposedUpdate.getArticle.updateComment,
      version: ProposedUpdate.getArticle.version,
    };

    return (
      <Container>
        <ScrollIndicator />
        <Header
          routeChangeAction={this.props.routeChangeAction}
          bgUpdated={
            proposed.attributes &&
            proposed.attributes.background &&
            current.attributes &&
            proposed.attributes.background !== current.attributes.background
          }
          date={proposed.dateCreated}
          oldTags={current.tags || []}
          newTags={proposed.tags || []}
          attributes={proposed.attributes}
          title={proposed.title}
          openModalAction={this.props.openModalAction}
          closeModalAction={this.props.closeModalAction}
          rejectArticleAction={this.props.rejectArticleAction}
          approveArticleAction={this.props.approveArticleAction}
          id={this.props.id}
          currentVersion={current.version}
          proposedVersion={proposed.version}
          contentHash={proposed.contentHash}
          author={proposed.author}
          owner={current.owner}
          communities={communities}
          currentUser={this.props.userId}
          status={proposed.status}
        />
        <Content>
          <Diffs current={current.content} proposed={proposed.content} />
          <Details>
            <Label>Update comment</Label>
            <p>
              {proposed.updateComment
                ? proposed.updateComment
                : "No update comment provided by the contributor"}
            </p>
          </Details>
        </Content>
      </Container>
    );
  }
}

export default ArticleReviewView;
