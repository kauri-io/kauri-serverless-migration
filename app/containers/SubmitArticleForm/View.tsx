import React from 'react'
import Form from 'antd/lib/form'
import SubmitArticleFormActions from './SubmitArticleFormActions'
import SubmitArticleFormHeader from './SubmitArticleFormHeader'
import SubmitArticleFormContent from './SubmitArticleFormContent'
import { showNotificationAction } from '../../lib/Epics/ShowNotificationEpic'
import AlertView from '../../components/Modal/AlertView'
import PublishingSelector, {
    IOption,
} from '../../containers/PublishingSelector'
import analytics from '../../lib/analytics'
import { path } from 'ramda'

import {
    IAttributesPayload,
    IEditArticlePayload,
    ISubmitArticlePayload,
    ISubmitArticleVersionPayload,
    IDraftArticleActionPayload,
} from './Module'

import { IPublishArticlePayload } from './PublishArticleModule'
import Loading from '../../components/Loading'

interface IOwner {
    id: string
    name: string
}

interface IProps {
    draftArticleAction: (payload: IDraftArticleActionPayload) => void
    submitArticleAction: (payload: ISubmitArticlePayload) => void
    submitArticleVersionAction: (payload: ISubmitArticleVersionPayload) => void
    editArticleAction: (payload: IEditArticlePayload) => void
    publishArticleAction: (payload: IPublishArticlePayload) => void
    id?: string
    data?: any
    article?: any
    form: any
    handleFormChange: any
    routeChangeAction: any
    username: string
    userId: string
    userAvatar: string
    showNotificationAction: typeof showNotificationAction
    openModalAction: (children?: any) => void
    closeModalAction: () => void
    communities: Array<{ community: IOption }>
}

interface ISubmitArticleVariables {
    subject: string
    text: string
    tags: string[]
    owner: IOwner
    version?: string
    attributes?: IAttributesPayload
}

class SubmitArticleForm extends React.Component<IProps> {
    static Header = SubmitArticleFormHeader
    static Actions = SubmitArticleFormActions
    static Content = SubmitArticleFormContent

    componentDidMount() {
        const {
            userId,
            routeChangeAction,
            communities,
            openModalAction,
            closeModalAction,
        } = this.props
        if (!userId) {
            routeChangeAction(
                `/login?r=${window.location.pathname}&redirected=true`
            )
        } else {
            analytics.track('Write Article Start', {
                category: 'generic',
            })
            if (
                communities &&
                communities.length > 0 &&
                !window.localStorage.getItem('community-publishing-modal')
            ) {
                openModalAction({
                    children: (
                        <AlertView
                            title="Publish to Community"
                            content={
                                <div>
                                    You can publish articles to your personal
                                    and also communities when selecting Publish
                                </div>
                            }
                            confirmButtonAction={() => {
                                window.localStorage.setItem(
                                    'community-publishing-modal',
                                    Date.now().toString()
                                )
                                closeModalAction()
                            }}
                            closeModalAction={() => {
                                window.localStorage.setItem(
                                    'community-publishing-modal',
                                    Date.now().toString()
                                )
                                closeModalAction()
                            }}
                            closeButtonText="Understood"
                            confirmButtonText="Learn More"
                        />
                    ),
                })
            }
        }
        analytics.track('Write Article Start', {
            category: 'generic',
        })
    }

    selectDestination() {
        const {
            openModalAction,
            closeModalAction,
            communities,
            userId,
            data,
        } = this.props

        // Updating article from a community I am in
        if (Number(path(['getArticle', 'version'])(data)) >= 1) {
            return this.handleSubmit('submit/update')(null)
        }
        // Submitting fresh article and I potentially want to choose a community?
        else if (communities && communities.length > 0) {
            return openModalAction({
                children: (
                    <PublishingSelector
                        userId={userId}
                        type="Articles"
                        closeModalAction={closeModalAction}
                        communities={communities.map(({ community }) => ({
                            ...community,
                            type: 'COMMUNITY',
                        }))}
                        handleSubmit={(destination, e) =>
                            this.handleSubmit(
                                'submit/update',
                                undefined,
                                destination
                            )(e)
                        }
                    />
                ),
            })
        } else {
            return this.handleSubmit('submit/update')(null)
        }
    }

    validateForm(next: any) {
        this.props.form.validateFieldsAndScroll(async (formErr: any) => {
            if (formErr) {
                return this.showFormError(formErr)
            } else {
                next()
            }
        })
    }

    showFormError = (formErr: any) => {
        Object.keys(formErr).map(errKey =>
            formErr[errKey].errors.map((err: { message: string }) =>
                this.props.showNotificationAction({
                    description: err.message,
                    message: 'Validation error!',
                    notificationType: 'error',
                })
            )
        )
        return console.error(formErr)
    }

    showGenericError = () => {
        return this.props.showNotificationAction({
            description:
                'Something went wrong with the article creation, please refresh the page and try again.',
            message: 'Editor Error',
            notificationType: 'error',
        })
    }

    handleSubmit = (
        submissionType: string,
        updateComment?: string,
        destination?: IOption
    ) => (e: React.SyntheticEvent<HTMLButtonElement> | null) => {
        console.log('*** SUBMITTING ***', updateComment)
        if (e) {
            e.preventDefault()
        }
        this.props.closeModalAction()
        this.props.form.validateFieldsAndScroll(
            async (
                formErr: any,
                { text, subject, attributes, tags }: ISubmitArticleVariables
            ) => {
                const {
                    submitArticleAction,
                    submitArticleVersionAction,
                    editArticleAction,
                    draftArticleAction,
                    userId,
                    communities,
                } = this.props

                if (formErr) {
                    return this.showFormError(formErr)
                }

                const articleData: any =
                    this.props.data && this.props.data.getArticle
                text = JSON.stringify({ markdown: text })

                // NEW DRAFT
                if (!articleData && submissionType === 'draft') {
                    return draftArticleAction({
                        attributes: attributes || {},
                        subject,
                        tags: tags || [],
                        text,
                    })
                }
                // NEW ARTICLE
                if (!articleData && submissionType === 'submit/update') {
                    return submitArticleAction({
                        attributes,
                        destination,
                        selfPublish: true,
                        subject,
                        tags,
                        text,
                    })
                }

                const { id, version, status, author, owner } = articleData

                if (!articleData.attributes) {
                    articleData.attributes = {}
                }

                switch (status) {
                    case 'PUBLISHED':
                        // OWNER'S DRAFT
                        if (
                            owner &&
                            userId === owner.id &&
                            submissionType === 'draft'
                        ) {
                            return submitArticleVersionAction({
                                attributes:
                                    (articleData.attributes && {
                                        ...articleData.attributes,
                                        ...attributes,
                                    }) ||
                                    articleData.attributes,
                                id,
                                subject,
                                tags: tags || articleData.tags,
                                text,
                            })
                            // OWNER'S UPDATE
                        } else if (
                            ((owner && userId === owner.id) ||
                                (Array.isArray(communities) &&
                                    communities.includes(owner.id))) &&
                            submissionType === 'submit/update'
                        ) {
                            return submitArticleVersionAction({
                                attributes:
                                    (attributes && {
                                        ...articleData.attributes,
                                        ...attributes,
                                    }) ||
                                    articleData.attributes,
                                id,
                                owner,
                                selfPublish: true,
                                subject,
                                tags: tags || articleData.tags,
                                text,
                            })
                            // CONTRIBUTORS' DRAFT
                        } else if (
                            owner &&
                            userId !== owner.id &&
                            submissionType === 'draft'
                        ) {
                            return submitArticleVersionAction({
                                attributes:
                                    (attributes && {
                                        ...articleData.attributes,
                                        ...attributes,
                                    }) ||
                                    articleData.attributes,
                                id,
                                owner,
                                subject,
                                tags: tags || articleData.tags,
                                text,
                            })
                            // CONTRIBUTORS' UPDATE
                        } else if (
                            owner &&
                            userId !== owner.id &&
                            submissionType === 'submit/update'
                        ) {
                            return submitArticleVersionAction({
                                attributes:
                                    (attributes && {
                                        ...articleData.attributes,
                                        ...attributes,
                                    }) ||
                                    articleData.attributes,
                                id,
                                owner,
                                selfPublish: false,
                                subject,
                                tags: tags || articleData.tags,
                                text,
                                updateComment,
                            })
                        } else {
                            return this.showGenericError()
                        }
                    case 'DRAFT':
                        if (
                            author &&
                            userId === author.id &&
                            submissionType === 'draft'
                        ) {
                            return editArticleAction({
                                attributes:
                                    (attributes && {
                                        ...articleData.attributes,
                                        ...attributes,
                                    }) ||
                                    articleData.attributes,
                                id,
                                subject,
                                tags: tags || articleData.tags,
                                text,
                                version,
                            })
                        } else if (
                            author &&
                            userId === author.id &&
                            submissionType === 'submit/update'
                        ) {
                            return editArticleAction({
                                attributes:
                                    (attributes && {
                                        ...articleData.attributes,
                                        ...attributes,
                                    }) ||
                                    articleData.attributes,
                                id,
                                selfPublish: true,
                                subject,
                                tags: tags || articleData.tags,
                                text,
                                version,
                            })
                        } else {
                            return this.showGenericError()
                        }
                    case 'PENDING':
                    // pending articles should not be shown in the editor
                    default:
                        return this.showGenericError()
                }
            }
        )
    }

    render() {
        const { routeChangeAction, userId } = this.props

        if (!userId) {
            return <Loading />
        }

        const articleData = this.props.data && this.props.data.getArticle
        return (
            <Form>
                <SubmitArticleForm.Actions
                    {...this.props.form}
                    handleSubmit={this.handleSubmit}
                    selectDestination={() =>
                        this.validateForm(() => this.selectDestination())
                    }
                    routeChangeAction={routeChangeAction}
                    text={articleData && articleData.content}
                    status={articleData && articleData.status}
                    userId={this.props.userId}
                    author={
                        articleData &&
                        articleData.author &&
                        articleData.author.id
                    }
                    owner={
                        articleData && articleData.owner && articleData.owner.id
                    }
                    openModalAction={this.props.openModalAction}
                    closeModalAction={this.props.closeModalAction}
                    showNotificationAction={this.props.showNotificationAction}
                    communities={this.props.communities.map(
                        ({ community }) => community.id
                    )}
                />
                <SubmitArticleForm.Header
                    {...this.props.form}
                    status={articleData && articleData.status}
                    subject={articleData && articleData.title}
                    attributes={articleData && articleData.attributes}
                    tags={articleData && articleData.tags}
                />

                <SubmitArticleForm.Content
                    {...this.props.form}
                    attributes={articleData && articleData.attributes}
                    openModalAction={this.props.openModalAction}
                    closeModalAction={this.props.closeModalAction}
                    id={articleData && articleData.id}
                    text={articleData && articleData.content}
                    ownerId={
                        articleData && articleData.owner && articleData.owner.id
                    }
                    username={
                        articleData && articleData.owner && articleData.owner.id
                            ? articleData &&
                              articleData.owner &&
                              articleData.owner.username
                            : (articleData &&
                                  articleData.author &&
                                  articleData.author.username) ||
                              this.props.username
                    }
                    userId={
                        (articleData &&
                            articleData.owner &&
                            articleData.owner.id) ||
                        this.props.userId
                    }
                    userAvatar={
                        articleData && articleData.owner && articleData.owner.id
                            ? articleData &&
                              articleData.owner &&
                              articleData.owner.avatar
                            : (articleData &&
                                  articleData.author &&
                                  articleData.author.avatar) ||
                              this.props.userAvatar
                    }
                />
            </Form>
        )
    }
}

const WrappedSubmitArticleForm = Form.create()(SubmitArticleForm)

export default WrappedSubmitArticleForm
