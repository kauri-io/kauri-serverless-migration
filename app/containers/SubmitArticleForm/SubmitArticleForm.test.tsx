import View from './View'
import { shallow } from 'enzyme'

describe('containers/SubmitArticleForm/View', () => {
    it('should match the snapshot', () => {
        const props = {}
        const wrapper = shallow(<View {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
