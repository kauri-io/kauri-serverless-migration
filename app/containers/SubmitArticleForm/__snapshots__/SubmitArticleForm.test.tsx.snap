// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/SubmitArticleForm/View should match the snapshot 1`] = `
<SubmitArticleForm
  form={
    Object {
      "getFieldDecorator": [Function],
      "getFieldError": [Function],
      "getFieldInstance": [Function],
      "getFieldProps": [Function],
      "getFieldValue": [Function],
      "getFieldsError": [Function],
      "getFieldsValue": [Function],
      "isFieldTouched": [Function],
      "isFieldValidating": [Function],
      "isFieldsTouched": [Function],
      "isFieldsValidating": [Function],
      "isSubmitting": [Function],
      "resetFields": [Function],
      "setFields": [Function],
      "setFieldsInitialValue": [Function],
      "setFieldsValue": [Function],
      "submit": [Function],
      "validateFields": [Function],
      "validateFieldsAndScroll": [Function],
    }
  }
/>
`;
